dali/pipeline/data/tensor.h

```diff
@@ -264,6 +264,38 @@ class Tensor : public Buffer<Backend> {
     ShareData(ptr, bytes, vector<Index>());
   }
 
+  /**
+   * @brief Wraps a TensorList and gives it a new shape
+   * TensorList has to be a valid tensor
+   * (there must be at least 1 tensor stored in TensorList,
+   * volumes of the new and old shape need to mach
+   * all tensors need to be stored without
+   * any offset between them)
+   */
+  inline void ShareDataReshape(TensorList<Backend> *tl, const vector<Index> &new_shape) {
+    DALI_ENFORCE(tl != nullptr, "Input TensorList is nullptr");
```

Comment:
Since it can't be null, maybe a reference?

Reply:
For the consistency with ShareData I would keep it as it is .